ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -92,15 +104,57 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDefinitions = append(containerDefinitions, containerDef)
 	}
 
+	var ecsFields *ECSFields
+	if ecsFieldsFileName != "" {
+		data, err := readECSFields(ecsFieldsFileName)
+		ecsFields = data
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var networkMode string
+	if ecsFields != nil {
+		networkMode = ecsFields.TaskDefinition.NetworkMode
+		// The task-role-arn flag should take precedence over a taskRoleArn value specified in ECS fields file
+		if taskRoleArn == "" {
+			taskRoleArn = ecsFields.TaskDefinition.TaskRoleArn
+		}
+	}
+
+	if networkMode == "" {
+		networkMode = "bridge"
+	}
+
 	taskDefinition := &ecs.TaskDefinition{
 		Family:               aws.String(taskDefinitionName),
 		ContainerDefinitions: containerDefinitions,
 		Volumes:              convertToECSVolumes(volumes),
 		TaskRoleArn:          aws.String(taskRoleArn),
+		NetworkMode:          aws.String(networkMode),
 	}
+
 	return taskDefinition, nil
 }
 
+func readECSFields(filename string) (*ECSFields, error) {
+	ecsFieldsData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		e := errors.Wrapf(err, "Error reading file '%v'", filename)
+		return nil, e
+	}
+
+	ecsFields := &ECSFields{}
+
+	err = goyaml.Unmarshal([]byte(ecsFieldsData), &ecsFields)
+	if err != nil {
+		e := errors.Wrapf(err, "Error unmarshalling yaml data from ECS fields")
+		return nil, e
```

Comment:
nit; can you directly return the error here and not assign it to a meaningless variable?  `return nil, errors.Wrapf(err, "Error unmarshalling yaml data from ECS fields")`.

Reply:
Oops, good point. (This was leftover from doing some other futzing with error formatting). Will fix!